src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp

```diff
@@ -428,13 +428,30 @@ p2ptest_set_or_cmp(char *boptr, size_t size, char pattern, bool set)
 {
   int stride = xrt_core::getpagesize();
 
+  // Generate the valid data vector
+  const size_t write_size = 1024;
+  char valid_data[write_size];
+  std::memset(valid_data, pattern, write_size);
+
   assert((size % stride) == 0);
   for (size_t i = 0; i < size; i += stride) {
-    if (set) {
-      boptr[i] = pattern;
-    }
-    else if (boptr[i] != pattern) {
-      return false;
+    // Due to test timeout we can only have one large write to test write combines
```

Comment:
Did you look into refactoring the test timeout code base?

Reply:
Performing only large writes of 1024 bytes would cause the test would take over eight minutes for a u280! I figured the better choice is to shorten the test.